File: app/src/main/java/com/test/xyz/demo/presentation/common/di/module/AppModule.java
package com.test.xyz.demo.presentation.common.di.module;

import android.app.Application;
import android.content.res.Resources;

import com.test.xyz.demo.presentation.common.di.DaggerApplication;
import com.test.xyz.demo.presentation.common.util.Logger;
import com.test.xyz.demo.presentation.common.util.LoggerProxy;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    DaggerApplication app;

    public AppModule(DaggerApplication application) {
        app = application;
    }

    @Provides
    @Singleton
    protected Application provideApplication() {
        return app;
    }

    @Provides
    @Singleton
    protected Resources provideResources() {
        return app.getResources();
    }

    @Provides
    @Singleton
    protected Logger provideLogger() {
        return new LoggerProxy();
    }
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/common/di/module/AppModule.kt
package com.test.xyz.demo.presentation.common.di.module

import android.app.Application
import android.content.res.Resources
import com.test.xyz.demo.presentation.common.di.DaggerApplication
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(internal var app: DaggerApplication) {

    @Provides
    @Singleton
    fun provideApplication(): Application {
        return app
    }

    @Provides
    @Singleton
    fun provideResources(): Resources {
        return app.resources
    }
}
